### PR TITLE
simpler template engine

### PR DESCRIPTION
**Pull request by cyrilou242**
- [core] improve json escape in template rendering
- [core] simpler template engine

### CODE REVIEW

File: thirdeye-server/src/test/resources/alertEvaluation.json [hunk 1]
```diff
@@ -99,7 +99,5 @@
         }
       ]
     }
-  },
-  "start": "${start}",
-  "end": "${end}"
+  }
```
Comment (suvodeep-pyne) [inline thirdeye-server/src/test/resources/alertEvaluation.json:102]: glitch? how does alert evaluation work without start and end?

**Review by suvodeep-pyne — APPROVED** [review]
LGTM

This is going deeper into the json serde / jackson libs. there are pros and cons with this. I think this should be ok since the problem at hand is precisely handling substitutions within json which needs to be escaped and is being handled implicitly which is somewhat opaque but nice.

**Comment by suvodeep-pyne** [comment]
Also, the StringTemplateUtils is probably a bad name at this point.

File: thirdeye-server/src/test/resources/alertEvaluation.json (hunk 1 above)
Comment (cyrilou242) [reply thirdeye-server/src/test/resources/alertEvaluation.json:102]: The `DetectionPipelineContext` passed to planExecutor$runPipeline contains a detection Interval.
It is consumed by the DataFetcher via the __timeFilterMacro but could be by other ways.

I did this change a long time ago now, I think the main pros at the time were:
- detectionInterval is required by almost all operators, and we must ensure it's the same.
- detectionInterval should be compatible with:
-- timezone
-- granularity bucket rouding
-- completeness delay
-- re-play (recent, for data mutability)

**Comment by cyrilou242** [comment]
> Also, the StringTemplateUtils is probably a bad name at this point.

agree, will rename it in another commit

**Pull request merged**